cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/DefaultScheduleBuilder.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright © 2017 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.internal.app.runtime.schedule;
+
+import co.cask.cdap.api.schedule.ConstraintProgramScheduleBuilder;
+import co.cask.cdap.api.schedule.ScheduleBuilder;
+import co.cask.cdap.internal.app.runtime.schedule.trigger.PartitionTrigger;
+import co.cask.cdap.internal.app.runtime.schedule.trigger.TimeTrigger;
+import co.cask.cdap.internal.schedule.ScheduleCreationSpec;
+import co.cask.cdap.proto.ProtoConstraint;
+import co.cask.cdap.proto.id.NamespaceId;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TimeZone;
+
+/**
+ * The default implementation of {@link ScheduleBuilder}.
+ */
+public class DefaultScheduleBuilder implements ConstraintProgramScheduleBuilder {
+
+  private final String name;
+  private final NamespaceId namespace;
+  private final String programName;
+  private final List<ProtoConstraint> constraints;
+  private String description;
+  private Map<String, String> properties;
+
+  public DefaultScheduleBuilder(String name, NamespaceId namespace, String programName) {
+    this.name = name;
+    this.description = "";
+    this.namespace = namespace;
+    this.programName = programName;
+    this.properties = new HashMap<>();
+    this.constraints = new ArrayList<>();
+  }
+
+  @Override
+  public ScheduleBuilder setDescription(String description) {
+    this.description = description;
+    return this;
+  }
+
+  @Override
+  public ScheduleBuilder setProperties(Map<String, String> properties) {
+    this.properties = ImmutableMap.copyOf(properties);
+    return this;
+  }
+
+  @Override
+  public ConstraintProgramScheduleBuilder withConcurrency(int max) {
+    // TODO: move this to validation
```

Comment:
oh, I'll remove this. the validation was already being done:
https://github.com/caskdata/cdap/blob/26a0411620980bf7c9088a52ba2d7483b848e890/cdap-proto/src/main/java/co/cask/cdap/proto/ProtoConstraint.java#L82-L82